checkbox-support/checkbox_support/parsers/xrandr.py

```diff
@@ -0,0 +1,91 @@
+# This file is part of Checkbox.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+"""
+This module offers a parser that takes the output of xrandr and parses it
+into a dictionary, where the keys are the output names and the values are
+lists of modes, where each mode is a named tuple containing the resolution,
+refresh rate, and whether the mode is preferred mode, and/or current mode.
+
+Considerations:
+The typical line from xrandr output looks like this:
+  3840x2160  144.00 +  120.00*  99.95   60.00
+
+If a mode is preferred, the `+` is appended to the refresh rate, like so:
+    3840x2160  144.00+  120.00*  99.95   60.00
+
+If a mode is the currently used mode, the `*` is appended, like so:
+    3840x2160  144.00   120.00*  99.95   60.00
+
+Given mode might be both, preferred and current, like so:
+    1920x1080  144.00+  120.00*  99.95   60.00
+
+The name of the output begins in the column 0, the next lines that inform
+about the modes are indented, like so:
+DP-3 AOC 2770M GDBFBHA000236
+    1920x1080  60.00*  59.94   59.93   50.00
+"""
+
+
+import re
+from collections import namedtuple, defaultdict
+
+# Define the namedtuple for mode information
+Mode = namedtuple(
+    "Mode", ["resolution", "refresh_rate", "is_preferred", "is_current"]
+)
+
+
+def parse_xrandr_line(line):
+    # Checking if it's a display information line
+    if "connected" in line:
+        # return the output name (e.g., eDP-1)
+        return "output", line.split()[0]
+    elif line.strip() and line[0] == " ":
+        # It's a mode line
+        mode_data = line.strip()
+        parts = mode_data.split()
+        resolution = parts[0]
+        # handle multiple refresh rates on the same line
+        refresh_rates = parts[1:]
+
+        modes = []
+        for rate in refresh_rates:
+            try:
+                refresh_rate = float(re.sub(r"[+*]", "", rate))
+            except ValueError:
+                # float couldn't be parsed
+                continue
+            is_preferred = "+" in rate
+            is_current = "*" in rate
+            modes.append(
+                Mode(resolution, refresh_rate, is_preferred, is_current)
+            )
+        return "mode", modes
+    return None
```

Comment:
Minor: this is sligly different to what we do above, which is a bit counterintuitive, and I would throw a ValueError like int or float! Also, should we log what line triggers this condition? so that in the log we can debug easily if this causes an issue down the line